src/env.cc

```diff
@@ -877,7 +877,9 @@ Environment::Environment(IsolateData* isolate_data,
     // The process shouldn't be able to neither
     // spawn/worker nor use addons or enable inspector
     // unless explicitly allowed by the user
-    options_->allow_native_addons = false;
+    if (!options_->allow_addons) {
+      options_->allow_native_addons = false;
```

Comment:
Instead of overloading the value used for `--addons`, can we just add this into `Environment::no_native_addons()`? Also it would be clearer if  `allow_addons` is renamed to something like `allow_addons_in_permissions` to differentiate this with `--addons`.

Reply:
I'm following a pattern we use on the permission model:

* `--allow-child-process`
* `--allow-worker`
* `--allow-fs-read`
* `--allow-fs-write`

I feel naming it as `--allow-addons-in-permissions` would be odd. The reason I'm not touching the `Environment::no_native_addons()` logic is that it would be less clear from a permission model perspective if we handle permissions in a separate point of nodejs initialization/getter. That's pretty much what I told Tobias.